executor/runner/runner.go

```diff
@@ -73,21 +73,22 @@ func New(ctx context.Context, m metrics.Reporter, logUploaders *uploader.Uploade
 	dockerRuntime := func(ctx context.Context, cfg config.Config) (runtimeTypes.Runtime, error) {
 		return docker.NewDockerRuntime(ctx, m, dockerCfg, cfg)
 	}
-	return WithRuntime(ctx, m, dockerRuntime, logUploaders, cfg, &dockerCfg)
+	return WithRuntime(ctx, m, dockerRuntime, logUploaders, cfg)
 }
 
 // WithRuntime builds an Executor using the provided Runtime factory func
-func WithRuntime(ctx context.Context, m metrics.Reporter, rp RuntimeProvider, logUploaders *uploader.Uploaders, cfg config.Config, dockerCfg *docker.Config) (*Runner, error) {
+func WithRuntime(ctx context.Context, m metrics.Reporter, rp RuntimeProvider, logUploaders *uploader.Uploaders, cfg config.Config) (*Runner, error) {
+	metricsTagger, _ := m.(tagger) // metrics.Reporter may or may not implement tagger interface.  OK to be nil
 	runner := &Runner{
-		logger:       logrus.NewEntry(logrus.StandardLogger()),
-		metrics:      m,
-		logUploaders: logUploaders,
-		config:       cfg,
-		dockerCfg:    dockerCfg,
-		taskChan:     make(chan task, 1),
-		killChan:     make(chan struct{}),
-		UpdatesChan:  make(chan Update, 10),
-		StoppedChan:  make(chan struct{}),
+		logger:        logrus.NewEntry(logrus.StandardLogger()),
+		metrics:       m,
+		metricsTagger: metricsTagger,
```

Comment:
Did these fields get reordered?

Reply:
no, I stuck in metricsTagger after metrics because they are closely related.  Other than that I did not reorder anything else.